openmetadata-airflow-apis/src/openmetadata/operations/last_dag_logs.py

```diff
@@ -0,0 +1,59 @@
+#  Copyright 2022 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Module containing the logic to retrieve all logs from the tasks of a last DAG run
+"""
+import glob
+import os
+from pathlib import Path
+
+from airflow.models import DagModel, DagRun
+from flask import Response
+from openmetadata.api.response import ApiResponse, ResponseFormat
+
+
+def last_dag_logs(dag_id: str) -> Response:
+    """
+    Validate that the DAG is registered by Airflow and have at least one Run.
+    If exists, returns all logs for each task instance of the last DAG run.
+    :param dag_id: DAG to find
+    :return: API Response
+    """
+
+    dag_model = DagModel.get_dagmodel(dag_id=dag_id)
+
+    if not dag_model:
+        return ApiResponse.not_found(f"DAG '{dag_id}' not found.")
+
+    last_dag_run = dag_model.get_last_dagrun()
+
+    if not last_dag_run:
+        return ApiResponse.not_found(f"No DAG run found for '{dag_id}'.")
+
+    task_instances = last_dag_run.get_task_instances()
+
+    response = {}
+
+    for task_instance in task_instances:
+        if os.path.isfile(task_instance.log_filepath):
+            response[task_instance.task_id] = Path(task_instance.log_filepath).read_text()
+        # logs could be kept in a directory with the same name than the log file path without extension per attempt
+        elif os.path.isdir(os.path.splitext(task_instance.log_filepath)[0]):
+            dir_path = os.path.splitext(task_instance.log_filepath)[0]
+            sorted_logs = sorted(filter(os.path.isfile, glob.glob(f"{dir_path}/*.log")), key=os.path.getmtime)
+            response[task_instance.task_id] = \
+                f"\n*** Reading local file: {task_instance.log_filepath}\n".join(
```

Comment:
@nahuelverdugo Thanks for the PR. Overall looks great.
In here, we will be reading all the logs for a task run. IMO we should take the run_id as a parameter give the log for that run itself. 
Example 
<img width="251" alt="image" src="https://user-images.githubusercontent.com/38649/170371603-de39df9e-d50e-4623-9c7c-f41300216184.png">
we want users to click one of those boxes, each box represent a run and get that specific log.

Reply:
Hi @harshach, wanted to simplify the first iteration by fetching the last logs also in case we wanted to provide a separated `Logs` button in Actions to jump directly to the running/last executed pipeline.

the status endpoint is already returning the run_id so we could add that lookup based on this logic

Reply:
make sense. However this call become slow and slow as more executions runs that means each run generates a new log file and when we make a request to fetch logs it will try to load more data

Reply:
oh yeah sorry, I did not mean to send the logs together with the status, just that the run_id info is already available as part of the status so adapting the flow should be easy

Reply:
Just to clarify a little bit this part of the code, each `DagRun` is composed of `TaskInstance`. The `TaskInstance` model has a property called `log_filepath` which is a path in the format `{AIRFLOW_HOME}/logs/{dag_id}/{task_id}/{execution_date}.log` but what we really have in the server is `{AIRFLOW_HOME}/{dag_id}/{task_id}/{execution_date}/{try_number}.log` (as long as we have not changed the logging configuration). 

Because of that, the code will check if the file of `log_filepath` exists just in case, otherwise, check if there are log files in the directory. Then, it joins all them together trying to do the same behaviour that you obtain when clicking in `All` logs of a Task Instance in Ariflow.

![image](https://user-images.githubusercontent.com/11074908/170435994-a2dc8120-e5dd-44b6-881b-5f54c1e03550.png)

It is true what @harshach says, it is going to be bigger but not every run, but every attempt.